In [1]:
import pandas as pd


In [2]:
# Creating the data_for positive samples 

FILE_PATH="metadata.csv"
IMAGES_PATH="covidimages"


df=pd.read_csv(FILE_PATH)
print(df.shape)

(758, 29)


In [3]:
df.head()

patientid offset sex   age   finding survival intubated intubation_present  \
0         2      0   M  65.0  COVID-19        Y         N                  N   
1         2      3   M  65.0  COVID-19        Y         N                  N   
2         2      5   M  65.0  COVID-19        Y         N                  N   
3         2      6   M  65.0  COVID-19        Y         N                  N   
4         4      0   F  52.0  COVID-19      NaN         N                  N   

  went_icu in_icu  ...              date  \
0        N      N  ...  January 22, 2020   
1        N      N  ...  January 25, 2020   
2        N      N  ...  January 27, 2020   
3        N      N  ...  January 28, 2020   
4        N      N  ...  January 25, 2020   

                                            location  folder  \
0        Cho Ray Hospital, Ho Chi Minh City, Vietnam  images   
1        Cho Ray Hospital, Ho Chi Minh City, Vietnam  images   
2        Cho Ray Hospital, Ho Chi Minh City, Vietnam  images   
3        Cho Ray Hospital, Ho Chi Minh City, Vietnam  images   
4  Changhua Christian Hospital, Changhua City, Ta...  images   

                                            filename                   doi  \
0  auntminnie-a-2020_01_28_23_51_6665_2020_01_28_...  10.1056/nejmc2001272   
1  auntminnie-b-2020_01_28_23_51_6665_2020_01_28_...  10.1056/nejmc2001272   
2  auntminnie-c-2020_01_28_23_51_6665_2020_01_28_...  10.1056/nejmc2001272   
3  auntminnie-d-2020_01_28_23_51_6665_2020_01_28_...  10.1056/nejmc2001272   
4                              nejmc2001573_f1a.jpeg  10.1056/NEJMc2001573   

                                                 url license  \
0  https://www.nejm.org/doi/full/10.1056/NEJMc200...     NaN   
1  https://www.nejm.org/doi/full/10.1056/NEJMc200...     NaN   
2  https://www.nejm.org/doi/full/10.1056/NEJMc200...     NaN   
3  https://www.nejm.org/doi/full/10.1056/NEJMc200...     NaN   
4  https://www.nejm.org/doi/full/10.1056/NEJMc200...     NaN   

                                      clinical_notes other_notes Unnamed: 28  
0  On January 22, 2020, a 65-year-old man with a ...         NaN         NaN  
1  On January 22, 2020, a 65-year-old man with a ...         NaN         NaN  
2  On January 22, 2020, a 65-year-old man with a ...         NaN         NaN  
3  On January 22, 2020, a 65-year-old man with a ...         NaN         NaN  
4   diffuse infiltrates in the bilateral lower lungs         NaN         NaN  

[5 rows x 29 columns]

## We can see that there is a coloumn named finding that denotes the disease i.e whether there is covid or not 

## Now we will make a sperate folder only for covid positive xrays


In [6]:
# os is use to iterate over folders and move file into that
import os
import shutil



# Making a folder name covid inside the dataset directory
TARGET_DIR="Dataset/Covid"


# IF this path does not exist then make it i.e make directory(mkdir)
if not os.path.exists(TARGET_DIR):
    os.mkdir(TARGET_DIR)
    print("covid folder created")

covid folder created


## Now our empty covid folder is created so we will pick images of covid dataset and place inside this folder

In [7]:
cnt=0

for(i,row) in df.iterrows():
    if row['finding']=="COVID-19":
        cnt+=1
print(cnt)

521


## i.e we have 521 images of covid positive xray

## The problem here is that some of the xrays have top view while some have side view and so on but we only need the xrays of the front view
## More technically we want that we want posterior -anterior view of xray i.e PA view

In [9]:
cnt=0

for(i,row) in df.iterrows():
    if row['finding']=="COVID-19" and row["view"]=="PA":
        filename=row["filename"]
        image_path=os.path.join(IMAGES_PATH,filename)
        image_copy_path=os.path.join(TARGET_DIR,filename)
        shutil.copy2(image_path,image_copy_path)
        print("Moving Image ",cnt)
        cnt+=1
print(cnt)

Moving Image  0
Moving Image  1
Moving Image  2
Moving Image  3
Moving Image  4
Moving Image  5
Moving Image  6
Moving Image  7
Moving Image  8
Moving Image  9
Moving Image  10
Moving Image  11
Moving Image  12
Moving Image  13
Moving Image  14
Moving Image  15
Moving Image  16
Moving Image  17
Moving Image  18
Moving Image  19
Moving Image  20
Moving Image  21
Moving Image  22
Moving Image  23
Moving Image  24
Moving Image  25
Moving Image  26
Moving Image  27
Moving Image  28
Moving Image  29
Moving Image  30
Moving Image  31
Moving Image  32
Moving Image  33
Moving Image  34
Moving Image  35
Moving Image  36
Moving Image  37
Moving Image  38
Moving Image  39
Moving Image  40
Moving Image  41
Moving Image  42
Moving Image  43
Moving Image  44
Moving Image  45
Moving Image  46
Moving Image  47
Moving Image  48
Moving Image  49
Moving Image  50
Moving Image  51
Moving Image  52
Moving Image  53
Moving Image  54
Moving Image  55
Moving Image  56
Moving Image  57
Moving Image  58
Moving 

## Now our dataset path i.e Dataset/Covid have images of covid x-ray patients
## i.e all the front view or posterior anterior view of xrays

## Now we have to do two things firstly we have to reduce the size of xray images so that our model becomes computational efficient and second we have to make  another folder for normal xrays

In [10]:
## Kaggle dataset is a big dataset with almost 6000 images of normal xray as well 
## but we want to have 50 -50 ratio of the covid and normal images hence we will do image sampling
## half covid images and half normal images


## Randomly choosing 201 (same as the covid xray images) images from the normal dataset

import random
KAGGLE_FILE_PATH="chest_xray/train/NORMAL"
TARGET_NORMAL_DIR="Dataset/Normal"


In [11]:
# Getting the names of all the images present in the file path
image_names=os.listdir(KAGGLE_FILE_PATH)


In [13]:
len(image_names)

1341

In [14]:
# Do the random shuffling of the image names 
random.shuffle(image_names)


for i in range(201):
    image_name=image_names[i]
    image_path=os.path.join(KAGGLE_FILE_PATH,image_name)
    target_path=os.path.join(TARGET_NORMAL_DIR,image_name)
    
    shutil.copy2(image_path,target_path)
    print("copying image ",i)
    
    
    
    
    
    
    

copying image  0
copying image  1
copying image  2
copying image  3
copying image  4
copying image  5
copying image  6
copying image  7
copying image  8
copying image  9
copying image  10
copying image  11
copying image  12
copying image  13
copying image  14
copying image  15
copying image  16
copying image  17
copying image  18
copying image  19
copying image  20
copying image  21
copying image  22
copying image  23
copying image  24
copying image  25
copying image  26
copying image  27
copying image  28
copying image  29
copying image  30
copying image  31
copying image  32
copying image  33
copying image  34
copying image  35
copying image  36
copying image  37
copying image  38
copying image  39
copying image  40
copying image  41
copying image  42
copying image  43
copying image  44
copying image  45
copying image  46
copying image  47
copying image  48
copying image  49
copying image  50
copying image  51
copying image  52
copying image  53
copying image  54
copying image  55
co

## Now our dataset is ready ie we have divided the dataset into the normal and covid cases 


## Since our image size is pretty large so we will use google colab to work on

## i.e data preprosessing we have done on our system rest of the work we will do on colab